tests/drivers/flash_simulator/src/main.c

```diff
@@ -5,7 +5,9 @@
  */
 
 #include <ztest.h>
+#ifdef CONFIG_FLASH
 #include <drivers/flash.h>
```

Comment:
This check seems strange, for flash_simulator I would think that CONFIG_FLASH cannot be disabled, so why check ?